File: problem/Min_Operations_Array/minOperations.py
class Solution:
    def minOperations(self, nums):
        def my_gcd(a, b):
            while b:
                a, b = b, a % b
            return a

        n = len(nums)
        ones = nums.count(1)
        if ones:
            return n - ones

        INF = 10**18
        res = INF

        for i in range(n):
            g = nums[i]
            for j in range(i + 1, n):
                g = my_gcd(g, nums[j])
                if g == 1:
                    res = min(res, j - i)
                    break

        if res == INF:
            return -1
        return res + n - 1

File: problem/Minimum_Changes/minOperations.c
int minOperations(char *s)
{
    int count = 0;
    int n = strlen(s);

    for (int i = 0; i < n; i++)
    {
        if (((s[i] - '0') ^ (i & 1)) & 1)
        {
            count++;
        }
    }

    int altCount = n - count;
    return (count < altCount) ? count : altCount;
}